### PR TITLE
TAL-20: Create Registration Tests

### PR DESCRIPTION
**Pull request by ElKordel**

### CODE REVIEW

File: src/main/java/org/example/pages/RegistrationPage.java [hunk 1]
```diff
@@ -11,6 +12,14 @@ public class RegistrationPage extends BasePage {
     private static final By PASSWORD_INPUT_LOCATOR = By.xpath("//input[@name='password1']");
     private static final By REPEAT_PASSWORD_INPUT_LOCATOR = By.xpath("//input[@name='password2']");
     private static final By REGISTRATION_BUTTON_LOCATOR = By.xpath("//button[contains(text(), 'Register User')]");
+    private static final By ALREADY_HAVE_AN_ACCOUNT_LOCATOR = By.xpath("//a[contains(text(), 'Login!')]");
+    private static final By CREATE_A_USER_TITLE_LOCATOR = By.xpath("//h1[text()='Create a User!']");
+    private static final By MORE_150_OVER_CHARACTERS_ERROR_LOCATOR = By.xpath("//li[text()='Ensure this value has at most 150 characters (it has 151).']");
+    private static final By FORBIDDEN_CHARACTERS_ERROR_LOCATOR = By.xpath("//li[text()='Enter a valid username. This value may contain only letters, numbers, and @/./+/-/_ characters.']");
+    private static final By PASSWORD_AND_REPEATPASSWORD_DIFFERENT_ERROR_LOCATOR = By.xpath("//li[text()='The two password fields didn’t match.']");
```
Comment (Manuyafox) [inline src/main/java/org/example/pages/RegistrationPage.java:19]: Maybe PASSWORDS instead of PASSWORD_AND_REPEATPASSWORD? (Or at least REPEAT_PASSWORD)

File: src/main/java/org/example/pages/RegistrationPage.java [hunk 2]
```diff
@@ -11,6 +12,14 @@ public class RegistrationPage extends BasePage {
...
+    private static final By SHORT_PASSWORD_ERROR_LOCATOR = By.xpath("//li[text()='This password is too short. It must contain at least 8 characters.']");
+    private static final By PASSWORD_ONLY_DIGITS_ERROR_LOCATOR = By.xpath("//li[text()='This password is entirely numeric.']");
+    private static final By PASSWORD_AND_USERNAME_ARE_SIMILAR_ERROR_LOCATOR = By.xpath("//li[text()='The password is too similar to the username.']");
```
Comment (Manuyafox) [inline src/main/java/org/example/pages/RegistrationPage.java:22]: What about CREDENTIALS instead of PASSWORD_AND_USERNAME?

File: src/main/java/org/example/pages/RegistrationPage.java [hunk 3]
```diff
@@ -36,4 +45,75 @@ public RegistrationPage clickRegistrationButton() {
...
+        Assertions.assertTrue(getDriver().findElement(SHORT_PASSWORD_ERROR_LOCATOR).isDisplayed());
+        return this;
+    }
+
+    public RegistrationPage checkPasswordOnlyDigitsError() {
+        Assertions.assertTrue(getDriver().findElement(PASSWORD_ONLY_DIGITS_ERROR_LOCATOR).isDisplayed());
+        return this;
+    }
+
+    public RegistrationPage checkPasswordAndUsernameAreSimilarError() {
```
Comment (Manuyafox) [inline src/main/java/org/example/pages/RegistrationPage.java:104]: What about Credentials instead of PasswordAndUsername?

File: src/test/java/puppy/registration/RegistrationDifferentPasswordAndRepeatPasswordTest.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package puppy.registration;
+
+import org.example.pages.RegistrationPage;
+import org.junit.jupiter.api.Test;
+import puppy.base.BaseTest;
+
+public class RegistrationDifferentPasswordAndRepeatPasswordTest extends BaseTest {
```
Comment (Manuyafox) [inline src/test/java/puppy/registration/RegistrationDifferentPasswordAndRepeatPasswordTest.java:7]: Maybe Passwords instead of PasswordAndRepeatPassword?

File: src/main/java/org/example/pages/RegistrationPage.java [hunk 5]
```diff
@@ -36,4 +45,75 @@ public RegistrationPage clickRegistrationButton() {
...
+        Assertions.assertTrue(getDriver().findElement(MORE_150_OVER_CHARACTERS_ERROR_LOCATOR).isDisplayed());
+        return this;
+    }
+
+    public RegistrationPage checkForbiddenCharactersError() {
+        Assertions.assertTrue(getDriver().findElement(FORBIDDEN_CHARACTERS_ERROR_LOCATOR).isDisplayed());
+        return this;
+    }
+
+    public RegistrationPage checkPasswordAndRepeatPasswordDiffError() {
```
Comment (Manuyafox) [inline src/main/java/org/example/pages/RegistrationPage.java:89]: Passwords?

File: src/test/java/puppy/registration/RegistrationEmptyPasswordFieldTest.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+    private static final String USERNAME = "Newaccount7";
+    private final RegistrationPage registrationPage = new RegistrationPage();
+
+    @Test
+    public void registrationEmptyPasswordFieldTest() {
+        registrationPage
+                .navigateToRegistrationPage()
+                .enterUsername(USERNAME)
+                .clickRegistrationButton()
```
Comment (Manuyafox) [inline src/test/java/puppy/registration/RegistrationEmptyPasswordFieldTest.java:17]: I think you can add a method to clear password fields here, like in the login test (empty credentials login test). Before click to registration button

File: src/test/java/puppy/registration/RegistrationEmptyUsernameFieldTest.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    private final RegistrationPage registrationPage = new RegistrationPage();
+
+
+    @Test
+    public void registrationEmptyUsernameFieldTest() {
+        registrationPage
+                .navigateToRegistrationPage()
+                .enterPassword(PASSWORD)
+                .enterRepeatPassword(PASSWORD)
+                .clickRegistrationButton()
```
Comment (Manuyafox) [inline src/test/java/puppy/registration/RegistrationEmptyUsernameFieldTest.java:19]: As in the previous test, you can add a username field clearing method here

File: src/test/java/puppy/registration/RegistrationRepeatPasswordFieldTest.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
+package puppy.registration;
+
+import org.example.pages.RegistrationPage;
+import org.junit.jupiter.api.Test;
+import puppy.base.BaseTest;
+
+public class RegistrationRepeatPasswordFieldTest extends BaseTest {
```
Comment (Manuyafox) [inline src/test/java/puppy/registration/RegistrationRepeatPasswordFieldTest.java:7]: Without the word "empty" it is not clear what the test is about

**Review by Manuyafox — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/pages/RegistrationPage.java [hunk 9]
```diff
@@ -36,4 +45,75 @@ public RegistrationPage clickRegistrationButton() {
...
+        Assertions.assertTrue(getDriver().findElement(PASSWORD_ONLY_DIGITS_ERROR_LOCATOR).isDisplayed());
+        return this;
+    }
+
+    public RegistrationPage checkPasswordAndUsernameAreSimilarError() {
+        Assertions.assertTrue(getDriver().findElement(PASSWORD_AND_USERNAME_ARE_SIMILAR_ERROR_LOCATOR).isDisplayed());
+        return this;
+    }
+
+    public RegistrationPage checkLayout() {
```
Comment (Manuyafox) [inline src/main/java/org/example/pages/RegistrationPage.java:109]: You could also add a check for empty fields here, like in login layout test.

File: src/main/java/org/example/pages/RegistrationPage.java (hunk 2 above)
Comment (iiddd) [reply src/main/java/org/example/pages/RegistrationPage.java:22]: I'd say credentials is a too broad term here. Current name shows what this error is about.

File: src/main/java/org/example/pages/RegistrationPage.java [hunk 10]
```diff
@@ -36,4 +45,75 @@ public RegistrationPage clickRegistrationButton() {
...
+        Assertions.assertTrue(getDriver().findElement(ALREADY_HAVE_AN_ACCOUNT_LOCATOR).isDisplayed());
+        return this;
+    }
+
+    public RegistrationPage checkCreateAUserTitleIsDisplayed() {
+        Assertions.assertTrue(getDriver().findElement(CREATE_A_USER_TITLE_LOCATOR).isDisplayed());
+        return this;
+    }
+
+    public RegistrationPage check150OverCharactersError() {
```
Comment (iiddd) [inline src/main/java/org/example/pages/RegistrationPage.java:79]: I'd call it TooLongValueError or something like that. It's generally considered as a bad practice to use digits in names.

File: src/test/java/puppy/registration/RegistrationUsernameALotOfCharactersTest.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
+package puppy.registration;
+
+import org.example.pages.RegistrationPage;
+import org.junit.jupiter.api.Test;
+import puppy.base.BaseTest;
+
+public class RegistrationUsernameALotOfCharactersTest extends BaseTest {
+
+    private static final String USERNAME = "AbobusgavkAbobusgavkAbobusgavkAbobusgavkAbobusgavkAbobusgavkAbobu" +
```
[the remark this answers is not in the document]
Comment (iiddd) [reply src/test/java/puppy/registration/RegistrationUsernameALotOfCharactersTest.java:9]: Well if you just need to reach some character count, it's a good example to use String generator instead of this constant.

**Review by Manuyafox — APPROVED** [review]
